File: doc/logback-socketAppender/demo/src/main/java/com/example/demo/DemoApplication.java
package com.example.demo;

import ch.qos.logback.classic.LoggerContext;
import ch.qos.logback.classic.net.SimpleSocketServer;
import org.slf4j.ILoggerFactory;
import org.slf4j.LoggerFactory;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class DemoApplication {

	public static void main(String[] args) {
		/*
		 * logback 远程日志服务端启动方式，启动服务端需要注释掉 logback.xml 中客户端代码
		 */
		// ILoggerFactory iLoggerFactory = LoggerFactory.getILoggerFactory();
		// SimpleSocketServer simpleSocketServer = new SimpleSocketServer((LoggerContext) iLoggerFactory, 8080);
		SpringApplication.run(DemoApplication.class, args);
		// simpleSocketServer.start();
	}

}

File: kotlin-learn/src/main/kotlin/com/feng/demo/DemoApplication.kt
package com.feng.demo

import ch.qos.logback.classic.LoggerContext
import ch.qos.logback.classic.net.SimpleSocketServer
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.core.env.Environment

@SpringBootApplication
class DemoApplication

// @Autowired
// lateinit var environment: Environment

fun main(args: Array<String>) {
	val factory = LoggerFactory.getILoggerFactory()
	val socketServer = SimpleSocketServer(factory as LoggerContext, 10004)
	runApplication<DemoApplication>(*args)
	// println(environment.getProperty("local.server.port"))
	socketServer.start()
}
